web/client/utils/CoordinatesUtils.js

```diff
@@ -254,15 +255,34 @@ const CoordinatesUtils = {
                 newExtent[2] = point[0] + point[0] * 0.01;
                 newExtent[3] = point[1] + point[1] * 0.01;
             }else if (geoJSON.type === "GeometryCollection") {
-                geoJSON.geometies.reduce((extent, geometry) => {
+                return geoJSON.geometies.reduce((extent, geometry) => {
                     let ext = this.getGeoJSONExtent(geometry);
                     if (this.isValidExtent(ext)) {
-                        extent[0] = (ext[0] < newExtent[0]) ? ext[0] : newExtent[0];
-                        extent[1] = (ext[1] < newExtent[1]) ? ext[1] : newExtent[1];
-                        extent[2] = (ext[2] > newExtent[2]) ? ext[2] : newExtent[2];
-                        extent[3] = (ext[3] > newExtent[3]) ? ext[3] : newExtent[3];
+                        return [
+                            (ext[0] < extent[0]) ? ext[0] : extent[0],
```

Comment:
Sure, as you see the original one had many side effects. I didn't noticed they become exactly the same function after my refactor.